anima/env/mayaEnv/auxiliary.py

```diff
@@ -851,25 +858,38 @@ def run_pre_publishers():
 
         # before running use the staging area to store the current version
         staging['version'] = version
-        run_publishers(type_name)
+        try:
+            run_publishers(type_name)
+        except (PublishError, RuntimeError) as e:
+            # do not forget to clean up the staging area
+            staging.clear()
+            # pop up a message box with the error
+            pm.confirmDialog(
+                title='SaveError',
+                icon='critical',
+                message='<b>%s</b><br/><br/>%s' % (
+                    'SCENE NOT SAVED!!!',
+                    str(''.join([i for i in stringify(e) if ord(i) < 128]))
```

Comment:
this is better than your previous code. But I don't really recall why we need to filter the error message. Will it not work if we directly use ``e.message`` or ``'%s' % e`` or directly ``e``.